src/core_plugins/kibana/public/context/components/size_picker/size_picker.js

```diff
@@ -0,0 +1,30 @@
+import _ from 'lodash';
+import uiModules from 'ui/modules';
+import contextSizePickerTemplate from './size_picker.html';
+import './size_picker.less';
+
+const module = uiModules.get('apps/context', [
+  'kibana',
+]);
+
+module.directive('contextSizePicker', function ContextSizePicker() {
+  return {
+    bindToController: true,
+    controller: ContextSizePickerController,
+    controllerAs: 'contextSizePicker',
+    replace: true,
+    restrict: 'E',
+    scope: {
+      count: '=',
+      setCount: '=',
+    },
+    template: contextSizePickerTemplate,
+  };
+});
+
+
+function ContextSizePickerController() {
+  this.getOrSetCount = (count) => (
+    _.isUndefined(count) ? this.count : this.setCount(count)
```

Comment:
Ah, now I get your meaning - sorry for being so dense. The documentation actually does not cover this case. It does not seem to be a problem in practice, but I agree it does not feel good to rely on that. I'll try to implement the double watch approach again. The first time I tried it felt really choppy when the input value was changed rapidly.